dvc/checkout.py

```diff
@@ -176,7 +177,12 @@ def _checkout_file(
 
 
 def _remove_redundant_files(path_info, fs, obj, cache, force):
-    existing_files = set(fs.walk_files(path_info))
+    if fs.scheme == Schemes.LOCAL and cache.repo:
+        existing_files = set(
+            cache.repo.dvcignore.walk_files(fs.walk(path_info))
+        )
+    else:
+        existing_files = set(fs.walk_files(path_info))
```

Comment:
Discusssed with @karajan1001 that we can move this check into `dvcignore` and do something like `dvcignore.walk_files(fs, path_info)` and let it ignore non-local filesystems